.github/scripts/test_trymerge.py

```diff
@@ -806,5 +806,143 @@ def test_dont_ignore_flaky_failures(self, *args: Any) -> None:
         )
 
 
+class TestGitHubPRGhstackDependencies(TestCase):
+    @mock.patch("trymerge.gh_get_pr_info", return_value=mock_gh_get_info())
+    def setUp(self, mock_gh_get_pr_info: mock.MagicMock) -> None:
+        self.pr = self.mock_pr(123, "Mock title", "Mock body text", closed=False)
+        pass
+
+    @staticmethod
+    def mock_pr(num: int, title: str, body: str, closed: bool) -> GitHubPR:
```

Comment:
I have a feeling that you don't need to devise your own mock here in `test_trymerge` unless there is a clear reason to do so.  The usual way to mock here is to use the generic patches below and uses actual PR in PyTorch:

```
@mock.patch("trymerge.get_rockset_results", side_effect=mocked_rockset_results)
@mock.patch("trymerge.gh_graphql", side_effect=mocked_gh_graphql)
```

Here is a recent example I have in mind https://github.com/pytorch/pytorch/pull/105098.  Before running the test, you can export `GITHUB_TOKEN` and `ROCKSET_API_KEY` readily.  When you run the test locally, `mocked_gh_graphql` would make a single query to PyTorch to get the PR info and cache them locally in `.github/scripts/gql_mocks.json`